src/extractor.ts

```diff
@@ -42,17 +42,53 @@ export class Extractor {
 	/**
 	 * Serialize and return output
 	 */
-	public serialize(): string {
-		return this.serializer.serialize(this.messages);
+	public serialize(messages: {[key: string]: string}): string {
+		return this.serializer.serialize(messages);
+	}
+
+	/**
+	 * Parse the content of a file and return output as an object
+	 */
+	public parse(contents: string): {[key: string]: string} {
+		return this.serializer.parse(contents);
 	}
 
 	/**
 	 * Serialize and save to destination
 	 */
-	public save(destination: string): string {
-		const data = this.serialize();
-		fs.writeFileSync(destination, data);
-		return data;
+	public save(destination: string, merge: boolean | 'clean' = true): string {
```

Comment:
What do you think of changing the signature for this method?

I think this is cleaner:

public save(destination: string, merge: boolean = true, clean: boolean = false): string